src/clusterfuzz/_internal/metrics/monitor.py

```diff
@@ -114,13 +115,21 @@ def run(self):
                    f'metric_kind: {series.metric_kind}, '
                    f'start_time: {metric_st_sec}.{metric_st_ns}, '
                    f'end_time: {metric_et_sec}.{metric_et_ns}')
-          time_series.append(series)
+
+          if (series.metric_kind == metric_pb2.MetricDescriptor.MetricKind.GAUGE  # pylint: disable=no-member
+             ):
+            gauge_series.append(series)
+          else:
+            time_series.append(series)
 
           if len(time_series) == MAX_TIME_SERIES_PER_CALL:
+            time_series.extend(gauge_series)
```

Comment:
IIUC this relies on the fact that gauge metric points are tagged with `start_time = end_time`, which makes them out of order with other metrics. However it seems possible to have time series points that have an even later start time, e.g.:

- time series X: start = 1, end = 2
- gauge series Y: start = 3, end = 3
- time series Z: start = 4, end = 5

Right?

It seems safer to me to sort the time series list in the way that the monitoring service expects it to be before sending it off.

Reply:
Done. Added Sorting based solution.